python/ray/serve/controller.py

```diff
@@ -714,8 +716,11 @@ def get_serve_instance_details(self) -> Dict:
         # route_prefix is set instead in each application.
         # Eventually we want to remove route_prefix from DeploymentSchema.
         return ServeInstanceDetails(
-            host=http_config.host,
-            port=http_config.port,
+            proxy_location=http_config.location,
+            http_options=HTTPOptionsSchema(
+                host=http_config.host,
+                port=http_config.port,
+            ),
```

Comment:
btw, this changed broke the frontend UI. 

The ServeInstanceDetails class and all of its fields are now public apis that power the frontend UI. They need to be changed in backwards compatible ways or the frontend has to be updated when these things change.

In this case, the frontend is not yet enabled publicly to users but this is a FYI for the future.

Reply:
We should have better automated testing once we start generating the web models from the API models.

Reply:
ah yeah sorry this was an intentional breakage... would be great to have some automated compatibility testing for the request & response schemas, maybe we can get that on the plan for next release